### PR TITLE
refactor referred gym flow using local storage

### PR DESCRIPTION
**Pull request by nerder**
I took inspiration for the `localStorage` api: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage

### CODE REVIEW

File: lib/src/blocs/dynamic_link/dynamic_link_bloc.dart [hunk 1]
```diff
@@ -30,8 +35,11 @@ class DynamicLinkBloc extends Bloc<DynamicLinkEvent, DynamicLinkState> {
       String path = event.deepLink.path;
 
       if (path.startsWith("/register/")) {
-        yield DynamicLinkSetDefaultGym(
-            defaultGym: path.replaceAll("/register/", ""));
+        final referredGymId = path.replaceAll("/register/", "");
+        await _localStorageRepository.setItem(
+          "referredGym",
+          referredGymId,
+        );
```
Comment (deathcoder) [inline lib/src/blocs/dynamic_link/dynamic_link_bloc.dart:42]: probably better to expose specific getter and setters in the localStorageRepository, something like _localStorageRepository.setReferredGym

this way the key remains an implementation detail

File: lib/src/blocs/login/login_bloc.dart [hunk 2]
```diff
@@ -105,4 +119,11 @@ class LoginBloc extends Bloc<LoginEvent, LoginState> {
       yield LoginSuccess(loggedUser: testOwner);
     }
   }
+
+  Future<dynamic> _getReferredGym() async {
```
Comment (deathcoder) [inline lib/src/blocs/login/login_bloc.dart:123]: this should be the correct type: Future<String>

**Review by deathcoder — COMMENTED** [review]
does this change remove the feature that allows users to change their currentGymId using a referral link? it seems to me now it only works in the createUser

**Comment by nerder** [comment]
It was, now I've add support for that, so that you can change gym at any point in time, it works in the pre-login phase or later on in the flow. 

I guess this solution can be applied for other implementations as well

**Pull request merged**